src/lib/WalletInfoCollector.ts

```diff
@@ -125,9 +125,15 @@ export default class WalletInfoCollector {
         let derivedAccountsPromise = WalletInfoCollector._deriveAccounts(startIndex,
             ACCOUNT_MAX_ALLOWED_ADDRESS_GAP, walletType, keyId);
 
+        // For ledger retrieve the keyId from the connected Ledger.
+        // Doing this after starting deriveAccounts, as the call is cheaper then.
+        if (walletType === WalletType.LEDGER) {
+            keyId = await LedgerApi.getWalletId();
+        }
+
         // Get or create the walletInfo instance and derive the first set of derived accounts
         const [walletInfo, firstSetOfDerivedAccounts] = await Promise.all([
-            WalletInfoCollector._getWalletInfoInstance(walletType, keyId),
+            WalletInfoCollector._getWalletInfoInstance(walletType, keyId!),
```

Comment:
Instead of using the `!notnull` operator here, I would prefer if the parameter in the function would have a default value instead of being nullable.

That way there is really no way that the keyId is empty, even when this method is used with the keyId omitted.